server/src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -116,6 +116,18 @@ public class IndicesClusterStateService extends AbstractLifecycleComponent imple
         Setting.Property.NodeScope
     );
 
+    /**
+     * Maximum number of shards to try and close concurrently. Defaults to the smaller of {@code node.processors} and {@code 10}, but can be
+     * set to any positive integer.
+     */
+    public static final Setting<Integer> CONCURRENT_SHARD_CLOSE_LIMIT = Setting.intSetting(
+        "indices.store.max_concurrent_closing_shards",
+        settings -> Integer.toString(Math.min(10, EsExecutors.NODE_PROCESSORS_SETTING.get(settings).roundUp())),
```

Comment:
Previously the default max was
```
final var maxThreads = Math.max(EsExecutors.NODE_PROCESSORS_SETTING.get(settings).roundUp(), 10);
```

Note `Math.max` instead of `Math.min`. Is this change intentional?

Reply:
Yes, it was a (my) mistake to use `max` here in the first place. I noticed the issue when I saw a small (IO-bound) node struggling to close lots of shards at once.